systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java

```diff
@@ -390,6 +394,79 @@ void testTlsExternalUser(ExtensionContext extensionContext) {
+        String entityOperatorPodName = kubeClient().listPodNamesInSpecificNamespace(namespace, Labels.STRIMZI_NAME_LABEL, KafkaResources.entityOperatorDeploymentName(userListeningClusterName)).get(0);
+        String uOLogs = kubeClient().logsInSpecificNamespace(namespace, entityOperatorPodName, "user-operator");
+        assertThat(uOLogs, containsString("KafkaUser " + testStorage.getUsername() + " in namespace " + namespace + " was ADDED"));
+
+        LOGGER.info("Verifying KafkaUser: {}/{} is not created in Kafka: {}/{}", namespace, testStorage.getUsername(), namespace, userIgnoringClusterName);
+        entityOperatorPodName = kubeClient().listPodNamesInSpecificNamespace(namespace, Labels.STRIMZI_NAME_LABEL, KafkaResources.entityOperatorDeploymentName(userIgnoringClusterName)).get(0);
+        uOLogs = kubeClient().logsInSpecificNamespace(namespace, entityOperatorPodName, "user-operator");
+        assertThat(uOLogs, not(containsString("KafkaUser " + testStorage.getUsername() + " in namespace " + namespace + " was ADDED")));
+
+        LOGGER.info("Verifying KafkaUser: {}/{} has label: {}={} corresponding to Kafka Cluster it belongs to", namespace, testStorage.getUsername(), Labels.STRIMZI_CLUSTER_LABEL, userListeningClusterName);
+        String kafkaUserResource = cmdKubeClient(namespace).getResourceAsYaml("kafkauser", testStorage.getUsername());
+        assertThat(kafkaUserResource, containsString(Labels.STRIMZI_CLUSTER_LABEL + ": " + userListeningClusterName));
+
+        LOGGER.info("Verifying that Secret corresponding to KafkaUser: {}/{} has expected label: {}={}", namespace, testStorage.getUsername(), Labels.STRIMZI_CLUSTER_LABEL, userListeningClusterName);
+        TestUtils.waitFor("Wait for the Secret {}: {}/{} to exist while also having label to corresponding Kafka cluster", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> kubeClient().listSecrets(namespace)
+                    .stream()
+                    .anyMatch(secret -> testStorage.getUsername().equals(secret.getMetadata().getName())
+                        && secret.getMetadata().getLabels().containsKey(Labels.STRIMZI_CLUSTER_LABEL)
+                        && secret.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL).equals(userListeningClusterName)
+                    )
+        );
+
+        LOGGER.info("Verifying KafkaUser: {}/{} is mapped into Kafka resource user in Kafka: {}/{}", namespace, testStorage.getUsername(), namespace, userListeningClusterName);
+        TestUtils.waitFor("Wait for the KafkaUser CR mapping into a Kafka user resource", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> {
+                String getUserResult = KafkaCmdClient.describeUserUsingPodCli(namespace, scraperPodName, KafkaResources.plainBootstrapAddress(userListeningClusterName), "CN=" + testStorage.getUsername());
+                return getUserResult.contains(testStorage.getUsername());
+            });
```

Comment:
Same with this one. Generally in our test cases, we tried to minimize such code. Could this be also somehow refactored into utils classes?